libs/libcommon/src/libcommon/orchestrator.py

```diff
@@ -728,6 +801,153 @@ def is_big(config: str) -> bool:
+        self.files_impacted_by_commit = self.get_impacted_files()
+        if self.files_impacted_by_commit - {
+            "README.md",
+            ".gitattributes",
+            ".gitignore",
+        }:  # TODO: maybe support .huggingface.yaml later
+            raise SmartUpdateImpossibleBecauseOfUpdatedFiles(", ".join(self.files_impacted_by_commit)[:1000])
+        self.updated_yaml_fields_in_dataset_card = self.get_updated_yaml_fields_in_dataset_card()
+        if "dataset_info" in self.updated_yaml_fields_in_dataset_card:
+            raise SmartUpdateImpossibleBecauseOfUpdatedYAMLField("dataset_info")
+        if "configs" in self.updated_yaml_fields_in_dataset_card:
+            raise SmartUpdateImpossibleBecauseOfUpdatedYAMLField("configs")
+        if "viewer" in self.updated_yaml_fields_in_dataset_card:
+            raise SmartUpdateImpossibleBecauseOfUpdatedYAMLField("viewer")
+        self.add_task(
+            UpdateRevisionOfDatasetCacheEntriesTask(
+                dataset=self.dataset, old_revision=self.old_revision, new_revision=self.revision
+            )
+        )
+        if self.storage_clients:
+            for storage_client in self.storage_clients:
+                self.add_task(
+                    UpdateRevisionOfDatasetStorageTask(
+                        dataset=self.dataset,
+                        old_revision=self.old_revision,
+                        new_revision=self.revision,
+                        storage_client=storage_client,
+                    )
+                )
+
+    def get_diff(self) -> str:
+        headers = build_hf_headers(token=self.hf_token, library_name="dataset-viewer")
+        resp = get_session().get(
+            self.hf_endpoint + f"/datasets/{self.dataset}/commit/{self.revision}.diff", timeout=10, headers=headers
+        )
+        resp.raise_for_status()
+        if not isinstance(resp.content, bytes):  # for mypy
+            raise RuntimeError(f"failed reading /datasets/{self.dataset}/commit/{self.revision}.diff")
+        return resp.content.decode("utf-8")
+
+    def get_impacted_files(self) -> set[str]:
+        return set(
+            line[len("--- a/") :]
+            for line in self.diff.split("\n")
+            if line.startswith("--- a/") or line.startswith("+++ b/")
+        )
+
+    def get_updated_yaml_fields_in_dataset_card(self) -> list[str]:
+        if "README.md" not in self.files_impacted_by_commit:
+            return []
+        try:
+            with Path(
+                hf_hub_download(
+                    self.dataset,
+                    "README.md",
+                    repo_type="dataset",
+                    token=self.hf_token,
+                    revision=self.revision,
+                    endpoint=self.hf_endpoint,
+                )
+            ).open(mode="r", newline="", encoding="utf-8") as f:
+                dataset_card_data_dict = DatasetCard(f.read()).data.to_dict()
+        except Exception:
```

Comment:
why do we hide the exceptions?

Reply:
sure, but I think we should let them raise, or raise any other error, instead of considering that the README YAML has not changed in a manner that could change the dataset viewer results. It would fallback to "normal" update, which is what I would expect.

Reply:
Well if there is an error, the readme is considered empty, which does fall back to normal update if it used to be valid and have some content

Reply:
I see your point though: if there was no metadata in the readme and the user adds malformed metadata we should raise an error instead of considering it to be an empty readme.

I'll raise an error then